src/Orleans/Serialization/BinaryFormatterSerializer.cs

```diff
@@ -58,7 +61,10 @@ public void Serialize(object item, ISerializationContext context, Type expectedT
                 return;
             }
 
-            var formatter = new BinaryFormatter();
+            var formatter = new BinaryFormatter
+            {
+                Context = new StreamingContext(StreamingContextStates.All, context)
```

Comment:
to allow GrainReference inside of BinaryFormatter serialized objects, we pass the context. Technically it's not needed while serializing, but I want to be consistent.